### ISSUE TITLE
Detected call of `lr_scheduler.step()` before `optimizer.step()`

### ISSUE THREAD

**Issue by NiAmL** [ThilinaRajapakse/pytorch-transformers-classification#21]
When I train a model here `global_step, tr_loss = train(train_dataset, model, tokenizer)` I get this warning:

> 0.767207/usr/local/lib/python3.6/dist-packages/torch/optim/lr_scheduler.py:82: UserWarning: Detected call of `lr_scheduler.step()` before `optimizer.step()`. In PyTorch 1.1.0 and later, you should call them in the opposite order: `optimizer.step()` before `lr_scheduler.step()`.  Failure to do this will result in PyTorch skipping the first value of the learning rate schedule.See more details at https://pytorch.org/docs/stable/optim.html#how-to-adjust-learning-rate
>   "https://pytorch.org/docs/stable/optim.html#how-to-adjust-learning-rate", UserWarning)

This is the screenshot, just in case helps (the warning is next to the epoch counter).

<img width="960" alt="Untitled" src="https://user-images.githubusercontent.com/50815888/66177667-61aa4400-e688-11e9-9642-5ae777aedd57.png">

**Comment by ThilinaRajapakse** [comment]
I pushed a fix to this yesterday. Updating to the latest version of the repo should fix it.

Alternatively, you can just put `optimizer.step()` above `scheduler.step()` in the `train` function.

**Comment by NiAmL** [comment]
oh ok sorry, I just started this morning with the experiment I was doing yesterday and I didn't check the repo updates yet

**Comment by ThilinaRajapakse** [comment]
No worries!
